types/src/front/assistant/actions/retrieval.ts

```diff
@@ -96,26 +94,15 @@ type ConnectorProviderDocumentType =
   | Exclude<ConnectorProvider, "webcrawler">
   | "document";
 
-const providerMap: Record<string, ConnectorProviderDocumentType> = {
-  "managed-slack": "slack",
-  "managed-notion": "notion",
-  "managed-google_drive": "google_drive",
-  "managed-github": "github",
-  "managed-confluence": "confluence",
-  "managed-microsoft": "microsoft",
-  "managed-intercom": "intercom",
-};
-
-const providerRegex = new RegExp(`^(${Object.keys(providerMap).join("|")})`);
-
 export function getProviderFromRetrievedDocument(
   document: RetrievalDocumentType
 ): ConnectorProviderDocumentType {
-  const match = document.dataSourceId.match(providerRegex);
-  if (match && match[1]) {
-    return providerMap[match[1]];
+  if (document.dataSourceView) {
+    if (document.dataSourceView.dataSource.connectorProvider === "webcrawler") {
```

Comment:
`nit`

We have a typeguard now: `isWebsite(document.dataSourceView.dataSource)`.

Reply:
not available here will move forward with this for now :+1: